src/pint/utils.py

```diff
@@ -2129,3 +2131,20 @@ def get_unit(parname):
 
     ac = AllComponents()
     return ac.param_to_unit(parname)
+
+
+def normalize_designmatrix(M, params):
```

Comment:
This needs a docstring, maybe with a bit of math in it explaining what is happening.

Reply:
Added.